### ISSUE TITLE
Is there a training code?

### ISSUE THREAD

**Issue by theotheo** [cheese-roll/light-anime-face-detector#1]
Hello! I want to train LFFD with my own data. So I looking for a training code.

**Comment by cheese-roll** [comment]
Sorry, my training code is in another machine right now, but I can give you a general idea. I used modified code from [here](https://github.com/YonghaoHe/A-Light-and-Fast-Face-Detector-for-Edge-Devices/tree/master/face_detection) for training.

If you want to train on your own data, I advise you to look into `data_provider_farm` directory. Modify (or create new) pickle_provider.py so that:
- `write()` (and `text_list_adapter.py`) can generate a pre-processed pickle file from your own dataset.
- `read_by_index()` returns a single data point from that file (see inside the initialize function how they normally load the pickle)

After that, select the configuration file from `config_farm` directory. Change the data / pre-trained model paths (and the provider import if you create your own) and run Python on that file and you should be good to go.

**Comment by theotheo** [comment]
Thanks for the clarifications!

A couple of questions:
1) Did you train from scratch? Or fine-tuned face detection weights?
2) How long did the training take?

**Comment by cheese-roll** [comment]
1. I used their pre-trained face detection weight from [here](https://github.com/YonghaoHe/A-Light-and-Fast-Face-Detector-for-Edge-Devices/blob/master/face_detection/saved_model/configuration_10_320_20L_5scales_v2/train_10_320_20L_5scales_v2_iter_1000000.params) to fine tune on my dataset.
2. This depends on the difficulty of the task / dataset (especially the one that objects have varied aspect ratios as LFFD doesn't handle that well). In my case, it only took around ~2 hours (1060 GTX) to get a "usable" model (loss starts to plateau but not really optimized).

**Comment by theotheo** [comment]
Thank you very much for your help!
